src/context/basket.js

```diff
@@ -1,21 +1,34 @@
 import React, { createContext, useState } from 'react'
+import _ from 'lodash'
+import ItemList from '../pages/home/items-list'
+import Item from '../pages/item'
+
 
 
 
 export const BasketContext = createContext({
-    basket: '',
+    basket: [],
     updateBasket: () => null
 })
 
+
+  
+
 const BasketProvider = ({children}) => {
-    const [basket, setBasket] = useState ('')
+    const [basket, setBasket] = useState ([])
+
+ 
+    const updateBasket = (ItemList) => {
+    const incrementer = _.find(basket, {id: ItemList.id})
+        {incrementer === undefined ? setBasket((prevState) => [...prevState, {id: ItemList.id, quantity: 1}]) : setBasket(() => [ {id: ItemList.id, quantity: incrementer.quantity +1}])}
```

Comment:
You don't need to do === undefined, the return value of find is the object or undefined and undefined is a false value
You can just swap the arguments around so that increment is before the `:`